sklearn/datasets/rcv1.py

```diff
@@ -124,16 +144,18 @@ def fetch_rcv1(data_home=None, subset='all', download_if_missing=True,
     # load data (X) and sample_id
     if download_if_missing and (not exists(samples_path) or
                                 not exists(sample_id_path)):
-        file_urls = ["%s_test_pt%d.dat.gz" % (URL, i) for i in range(4)]
-        file_urls.append("%s_train.dat.gz" % URL)
         files = []
-        for file_url in file_urls:
+        for file_name, file_url in zip(FILE_NAMES, FILE_URLS):
```

Comment:
If you already have zip, why not zip the checksums in too, rather than using a different data structure?